### DISCUSSION TITLE
Crash after importing 12k Beatmapsets and opening Song Select

### DISCUSSION THREAD

**Discussion by ackhack** [ppy/osu#15558] [Q&A]
I imported all my stable Beatmaps and the game crashed after trying to open Song Select.
The Error Message said something like "A new guard page for the stack cannot be created".
I have 48Gb of RAM and the original SongsFolder had 112Gb, so maybe that was the Issue.
Didnt happen again after starting lazer.

Logs:
[performance.log](https://github.com/ppy/osu/files/7513844/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7513845/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7513846/updater.log)
[database.log](https://github.com/ppy/osu/files/7513847/database.log)
[network.log](https://github.com/ppy/osu/files/7513848/network.log)

Lazer-version: 
2021.1108.0

**Comment by bdach** [comment]
The error message strongly suggests memory exhaustion, but for what reasons, I am not able to say from the logs alone. Please observe your game's memory usage and let us know whether it gets too big to be considered reasonable at any point (anything past a few gigabytes I would call unreasonable).

We may ask you to submit a memory dump privately at a later stage if it is confirmed that the game is using too much memory. Please **DO NOT** post memory dumps publically here.

**Reply by peppy** [reply]
runtime.log is full of broken beatmap parsing, suggesting either a very long play session or happening during import.

did this happen after running the import? does it happen from a fresh startup? if it's just the import then that's an EF thing and is known. just restart osu! once.

**Reply by ackhack** [reply]
The Memory doesnt spike above 1.1Gb anymore while scrolling around in Song Select.

Yes it happened after importing the Beatmaps. It imported around 8k Maps between the last working opening of Song Select and the crashing one. I was playing while the import was happening and song select always needed some time to open, so i decided to let it run and did something else.

No it does not happen from a fresh startup.

Trying to open a Beatmap that threw an error while importing still throws the same error. See: [runtime.log](https://github.com/ppy/osu/files/7514966/runtime.log)

**Reply by peppy** [reply]
The beatmap in question is failing on hit objects of type 128 (not immediately sure how this can happen, @smoogipoo probably knows the types better than i do).

The import issue will be resolved after the migration to realm (a different database backing) has been completed. This is an ongoing effort.

**Reply by smoogipoo** [reply]
Type 128 is a "hold" type which is only valid in mania. The map seems to crash in osu!stable according to a comment (https://osu.ppy.sh/comments/794650). It's a non-fatal error in lazer's case.

Likely the reason for the crash is that those log lines are pasted to the notification overlay - I wouldn't be surprised if that's causing some memory/VRAM issues.
